code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm

```diff
@@ -96,21 +96,27 @@
 			sleep(6)
 			if(animation)
 				qdel(animation)
-		notransform = 0
-	return 1
+		notransform = FALSE
+	return TRUE
 
 /obj/item/bloodcrawl
 	name = "blood crawl"
 	desc = "You are unable to hold anything while in this form."
 	icon = 'icons/effects/blood.dmi'
 	flags = NODROP|ABSTRACT
 
+/obj/effect/temp_visual/bloodcrawl
+	icon_state = "bloodsparkles"
+	duration = 20
+
 /mob/living/proc/phasein(var/obj/effect/decal/cleanable/B)
 
 	if(notransform)
 		to_chat(src, "<span class='warning'>Finish eating first!</span>")
-		return 0
+		return FALSE
 	B.visible_message("<span class='warning'>[B] starts to bubble...</span>")
+	var/obj/effect/temp_visual/uniquebloodcrawl = new /obj/effect/temp_visual/bloodcrawl(B.loc)
+	uniquebloodcrawl.color = B.color
 	if(!do_after(src, 20, target = B))
 		return
```

Comment:
```suggestion
		qdel(uniquebloodcrawl)
		return
```
Currently the effect will stay even if the demon cancels the phasing or when the blood gets removed.